sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,436 @@
+
+    def __eq__(self, other: object) -> bool:
+        """Checks equality.
+
+        :param object other: Another object to compare with this instance. Currently, only comparisons with
+            `KeyVaultRSAPrivateKey` or `JsonWebKey` instances are supported.
+
+        :returns: True if the objects are equal; False otherwise.
+        :rtype: bool
+        """
+        if isinstance(other, KeyVaultRSAPublicKey):
+            return all(getattr(self._key, field) == getattr(other._key, field) for field in self._key._FIELDS)
+        if isinstance(other, JsonWebKey):
+            return all(getattr(self._key, field) == getattr(other, field) for field in self._key._FIELDS)
+        return False
+
+    def verifier(  # pylint:disable=docstring-missing-param,docstring-missing-return,docstring-missing-rtype
+        self, signature: bytes, padding: AsymmetricPadding, algorithm: HashAlgorithm
+    ) -> NoReturn:
+        """Not implemented. This method was deprecated in `cryptography` 2.0 and removed in 37.0.0."""
+        raise NotImplementedError()
+
+
+class KeyVaultRSAPrivateKey(RSAPrivateKey):
+    """An `RSAPrivateKey` implementation based on a key managed by Key Vault.
+
+    Only synchronous clients and operations are supported at this time.
+    """
+
+    def __init__(self, client: "CryptographyClient", key_material: JsonWebKey) -> None:
+        """Creates a `KeyVaultRSAPrivateKey` from a `CryptographyClient` and key.
+
+        :param client: The client that will be used to communicate with Key Vault.
+        :type client: :class:`~azure.keyvault.keys.crypto.CryptographyClient`
+        :param key_material: They Key Vault key's material, as a `JsonWebKey`.
+        :type key_material: :class:`~azure.keyvault.keys.JsonWebKey`
+        """
+        self._client: "CryptographyClient" = client
+        self._key: JsonWebKey = key_material
+
+    def decrypt(self, ciphertext: bytes, padding: AsymmetricPadding) -> bytes:
+        """Decrypts the provided ciphertext.
+
+        :param bytes ciphertext: Encrypted bytes to decrypt.
+        :param padding: The padding to use. Supported paddings are `OAEP` and `PKCS1v15`. For `OAEP` padding, supported
+            hash algorithms are `SHA1` and `SHA256`. The only supported mask generation function is `MGF1`. See
+            https://learn.microsoft.com/azure/key-vault/keys/about-keys-details for details.
+        :type padding: :class:`~cryptography.hazmat.primitives.asymmetric.padding.AsymmetricPadding`
+
+        :returns: The decrypted plaintext, as bytes.
+        :rtype: bytes
+        """
+        mapped_algorithm = get_encryption_algorithm(padding)
+        result = self._client.decrypt(mapped_algorithm, ciphertext)
+        return result.plaintext
+
+    @property
+    def key_size(self) -> int:
+        """The bit length of the public modulus.
+
+        :returns: The key's size.
+        :rtype: int
+        """
+        private_key = self.private_numbers().private_key()
```

Comment:
The comment below from Scott inidiciates that calling `self.private_numbers()` us likely to fail typical usage.
It would be weird to get a `ValueError` while accessing a property - is there a more reliable way we can return this property?

Alternatively, if it's not possible, perhaps we should catch the error from `private_numbers` and tailor the message to this property.

Reply:
Thanks for the catch! I'll double-check for a more direct way to return the size, but otherwise I can handle the error here like we did in `private_bytes`